sql/sp.cc

```diff
@@ -964,7 +964,7 @@ enum_sp_return_code sp_drop_routine(THD *thd, enum_sp_type type,
     return SP_INTERNAL_ERROR;
 
 #ifdef WITH_WSREP
-  if (WSREP(thd) && wsrep_to_isolation_begin(thd, WSREP_MYSQL_DB, NULL, NULL)) {
+  if (WSREP(thd) && wsrep_to_isolation_begin(thd, name->m_db.str, NULL, NULL)) {
```

Comment:
Why this change?

Reply:
In order to check if the schema is read only, we need to first know the DB name. Here,  we were sending "mysql" as the DB name for all TOI operations on stored procedures and functions. This resulted in inconsistency when we execute create/drop operations on procedures and functions. So this change was required.